pkg/controller/certificaterequests/sync.go

```diff
@@ -167,6 +149,11 @@ func (c *Controller) setCertificateRequestStatus(cr *v1alpha1.CertificateRequest
 	// invalid cert
 	_, err := pki.DecodeX509CertificateBytes(cr.Status.Certificate)
 	if err != nil {
+		if cr.Status.FailureTime == nil {
+			nowTime := metav1.NewTime(c.clock.Now())
+			cr.Status.FailureTime = &nowTime
+		}
+
 		apiutil.SetCertificateRequestCondition(cr, v1alpha1.CertificateRequestConditionReady,
```

Comment:
Shouldn't both this line and the new code that you just added above be replaced with a call to the reporter?